src/instructlab/model/evaluate.py

```diff
@@ -213,29 +220,32 @@ def qa_pairs_to_qna_to_avg_scores(qa_pairs: list[dict]) -> dict:
     return qna_to_avg_scores
 
 
-def launch_server(
-    ctx,
-    model,
-    model_name,
-    max_workers,
-) -> tuple:
+def launch_server(ctx, model, model_name, max_workers, backend, max_ctx_size) -> tuple:
     # pylint: disable=import-outside-toplevel
     # First Party
-    from instructlab.model.backends import backends
-
-    if not ctx.obj.config.serve.backend:
-        ctx.obj.config.serve.backend = backends.VLLM
-    if ctx.obj.config.serve.backend == backends.VLLM:
-        ctx.obj.config.serve.vllm.vllm_args.extend(["--served-model-name", model_name])
-    elif ctx.obj.config.serve.backend == backends.LLAMA_CPP:
-        # mt_bench requires a larger context size
-        ctx.obj.config.serve.llama_cpp.max_ctx_size = 5120
+
+    # we shouldn't be in the business of editing a user's cfg.
+    # with eval, this is tricky since there are a few things we want to help them out with if they do not exist.
+    # if the user did not set a backend, or model_path, set that for them. Everything else needs to be in the cfg. or in the cmd as flag overrides.
+    temp_serve = copy.copy(ctx.obj.config.serve)
+    if backend is None:
+        temp_serve.backend = backends.VLLM
+    if backend == backends.VLLM:
+        temp_serve.vllm.vllm_args.extend(["--served-model-name", model_name])
+        temp_serve.vllm.max_ctx_size = max_ctx_size
+    if backend == backends.LLAMA_CPP:
+        if max_ctx_size != 5120:
+            click.secho(
+                f"Max ctx size of {max_ctx_size} does not match the requirement for the serving backend. For vllm the ctx_size should be 4096 and for llama-cpp it should be 5120. Please edit your config or use the --max-ctx-size flag.",
```

Comment:
We aren't setting the context size for vllm.  If we were, it would need to be the same as llama-cpp.